samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/opengl/UltraHDRWithOpenGL.kt

```diff
@@ -0,0 +1,269 @@
+                binding.openglBrightness100.isEnabled = true
+            }
+
+            else -> {
+                binding.openglBrightness30.isEnabled = false
+                binding.openglBrightness70.isEnabled = false
+                binding.openglBrightness100.isEnabled = false
+            }
+        }
+    }
+
+    private fun initUltraHDRGlRenderer() {
+        ultraHDRGLRenderer = UltraHDRWithOpenGLRenderer(requireContext(), bitmap)
+        desiredRatio = ultraHDRGLRenderer.desiredHdrSdrRatio
+        binding.imageContainer.setImageBitmap(bitmap)
+    }
+
+    private fun initGlRenderer() {
+        glRenderer = GLRenderer().apply {
+            registerEGLContextCallback(this@UltraHDRWithOpenGL)
+            start()
+        }
+    }
+
+    private fun initGlFrameRenderer() {
+        glFrameBufferRenderer = GLFrameBufferRenderer.Builder(binding.surfaceView, this)
+            .setGLRenderer(glRenderer)
+            .build()
+    }
+
+    override fun onAttach(context: Context) {
+        requireActivity().display
+            ?.registerHdrSdrRatioChangedListener(Runnable::run, updateHdrSdrRatio)
+        super.onAttach(context)
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        binding.colorModeControls.detach()
+        requireActivity().display
+            ?.unregisterHdrSdrRatioChangedListener(updateHdrSdrRatio)
+    }
+
+    override fun onDestroyView() {
+        glFrameBufferRenderer.release(true)
+        super.onDestroyView()
+    }
+
+    override fun onEGLContextCreated(eglManager: EGLManager) {
+        val colorSpaceName = when (isWideGamut) {
+            true -> ColorSpace.Named.DISPLAY_P3
+            false -> ColorSpace.Named.SRGB
+        }
+
+        ultraHDRGLRenderer.onContextCreated(ColorSpace.get(colorSpaceName))
+    }
+
+    override fun onEGLContextDestroyed(eglManager: EGLManager) {
+        ultraHDRGLRenderer.onContextDestroyed()
+    }
+
+    override fun onDrawFrame(
+        eglManager: EGLManager,
+        width: Int,
+        height: Int,
+        bufferInfo: BufferInfo,
+        transform: FloatArray,
+    ) {
+        hdrSdrRatio = requireActivity().display?.hdrSdrRatio ?: 1.0f
+        ultraHDRGLRenderer.onDrawFrame(
+            width,
+            height,
+            hdrSdrRatio,
+            bufferInfo.width,
+            bufferInfo.height,
+            transform,
+        )
+    }
+
+    override fun onDrawComplete(
+        targetSurfaceControl: SurfaceControlCompat,
+        transaction: SurfaceControlCompat.Transaction,
+        frameBuffer: FrameBuffer,
+        syncFence: SyncFenceCompat?,
+    ) {
+        transaction.setDataSpace(targetSurfaceControl, if (isWideGamut) P3_XRB else SRGB_XRB)
+        transaction.setExtendedRangeBrightness(targetSurfaceControl, hdrSdrRatio, desiredRatio)
+        transaction.commit()
+        super.onDrawComplete(targetSurfaceControl, transaction, frameBuffer, syncFence)
```

Comment:
Just FYI, delegation to the base class method is not necessary.

Also the implementation of GLFrameBufferRenderer already calls transaction.commit on your behalf so explicitly calling commit here is not necessary.

Reply:
👍🏾 Perfect - Thanks for letting me know